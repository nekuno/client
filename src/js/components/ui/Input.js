import React, { PropTypes, Component } from 'react';

export default class Input extends Component {

    static propTypes = {
        placeholder : PropTypes.string.isRequired,
        type        : PropTypes.string,
        defaultValue: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
        style       : PropTypes.object,
        doNotFocus  : PropTypes.bool,
        onChange    : PropTypes.func,
        onKeyDown   : PropTypes.func,
        maxLength   : PropTypes.string
    };

    constructor() {
        super();

        this.onFocusHandler = this.onFocusHandler.bind(this);
    }

    componentDidMount() {
        if (!this.props.doNotFocus) {
            this.focus();
        }
    }

    getValue() {
        return this.refs.input.value;
    }

    focus() {
        this.refs.input.focus();
    }

    render() {
        const {placeholder, type, defaultValue, style, maxLength, onChange, onKeyDown} = this.props;
        return (
            <li>
                <div className="item-content">
                    <div className="item-inner">
                        <div className="item-input">
                            <input ref="input" type={type || "text"}
                                   placeholder={placeholder}
                                   defaultValue={defaultValue}
                                   style={style}
                                   onKeyDown={onKeyDown}
                                   onChange={onChange}
                                   onFocus={this.onFocusHandler}
                                   maxLength={maxLength}/>
                        </div>
                    </div>
                </div>
            </li>
        );
    }

    onFocusHandler() {
        let inputElem = this.refs.input;
        if (inputElem) {
            window.setTimeout(function () {
                inputElem.scrollIntoView();
                document.getElementsByClassName('view')[0].scrollTop -= 100;
            }, 500);
        }
    }
}