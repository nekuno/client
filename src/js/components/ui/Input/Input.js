import PropTypes from 'prop-types';
import React, { Component } from 'react';
import styles from './Input.scss';

export default class Input extends Component {
    static propTypes = {
        placeholder : PropTypes.string,
        type        : PropTypes.string,
        defaultValue: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
        checked     : PropTypes.bool,
        doNotFocus  : PropTypes.bool,
        doNotScroll : PropTypes.bool,
        onChange    : PropTypes.func,
        maxLength   : PropTypes.string,
        searchIcon  : PropTypes.bool,
        size        : PropTypes.oneOf(['regular', 'small'])
    };

    constructor(props) {
        super(props);

        this.onChange = this.onChange.bind(this);
        this.onFocusHandler = this.onFocusHandler.bind(this);
        this.getValue = this.getValue.bind(this);
        this.clearValue = this.clearValue.bind(this);

        this.state = {
            empty: !props.defaultValue
        };
    }

    componentDidMount() {
        if (!this.props.doNotFocus) {
            this.focus();
        }
    }

    getValue() {
        return this.refs.input ? this.refs.input.value : null;
    }

    clearValue() {
        this.refs.input.value = '';
        this.onChange();
    }

    focus() {
        if (this.refs.input) {
            this.refs.input.focus();
        }
    }

    onFocusHandler() {
        const {doNotScroll} = this.props;
        let inputElem = this.refs.input;
        if (inputElem && !doNotScroll) {
            window.setTimeout(function () {
                inputElem.scrollIntoView();
                const viewClass = document.getElementsByClassName('view');
                if (viewClass && viewClass[0]) {
                    viewClass[0].scrollTop -= 100;
                }
            }, 500);
        }
    }

    onChange() {
        if (this.props.onChange) {
            this.props.onChange(this.getValue());
        }

        this.setState({empty: !this.getValue()});
    }

    render() {
        const {placeholder, type, defaultValue, checked, maxLength, searchIcon, size} = this.props;
        const {empty} = this.state;
        const sizeClass = size === 'small' ? styles.small : null;
        return (
            <div className={styles.inputWrapper + ' ' + sizeClass}>
                <div className={styles.input}>
                    {searchIcon && empty ?
                        <span className={styles.searchIcon + ' icon icon-search'}/>
                        : null
                    }
                    <input ref="input"
                           type={type || "text"}
                           placeholder={placeholder}
                           defaultValue={defaultValue}
                           onChange={this.onChange}
                           onFocus={this.onFocusHandler}
                           maxLength={maxLength}
                           required
                    />
                    {!empty && !checked ?
                        <span className={styles.iconCancel + ' ' + 'icon icon-x'} onClick={this.clearValue}/>
                        : null
                    }
                    {!empty && checked ?
                        <span className={'icon icon-check-circle ' + styles.iconCheck}/>
                        : null
                    }
                </div>
            </div>
        );
    }
}