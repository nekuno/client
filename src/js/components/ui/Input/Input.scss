@import '../../../../scss/_partials/colors';

.input-wrapper {
  padding: 16px;

  .input {
    background: $white-background;
    border-radius: 32px;
    color: $gray-main-text;
    height: 60px;
    line-height: 60px;
    margin: 16px;
    text-align: center;

    .search-icon {
      color: $gray-secondary-text;
      font-size: 26px;
      line-height: 60px;
    }

    input {
      border: none;
      font-size: 19px;
      height: 58px;
      width: 70%;
      margin: 0 5px;
      max-width: 80%;
      text-align: center;
      text-overflow: ellipsis;
      vertical-align: top;

      &:focus {
        outline: none;
      }
    }

    .icon-cancel {
      cursor: pointer;
    }
    .icon-check {
      color: $green-experiences
    }
  }

  &.small {
    padding: 8px;

    .input {
      border: 1px solid $gray-secondary-text;
      height: 40px;
      line-height: 40px;

      .search-icon {
        line-height: 40px;
      }

      input {
        font-size: 16px;
        height: 38px;
        text-align: left;
        line-height: 38px;
        width: 85%;
      }
    }
  }
}