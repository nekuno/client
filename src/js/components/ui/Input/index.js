import PropTypes from 'prop-types';
import React, { Component } from 'react';
import styles from './Input.scss';

export default class Input extends Component {
    static propTypes = {
        placeholder : PropTypes.string,
        type        : PropTypes.string,
        defaultValue: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
        checked     : PropTypes.bool,
        doNotFocus  : PropTypes.bool,
        doNotScroll : PropTypes.bool,
        onChange    : PropTypes.func,
        maxLength   : PropTypes.string,
        maxNum      : PropTypes.number,
        minNum      : PropTypes.number,
        searchIcon  : PropTypes.bool,
        textColor   : PropTypes.string,
        size        : PropTypes.oneOf(['regular', 'small']),
    };

    constructor(props) {
        super(props);

        this.onChange = this.onChange.bind(this);
        this.onFocusHandler = this.onFocusHandler.bind(this);
        this.onBlurHandler = this.onBlurHandler.bind(this);
        this.getValue = this.getValue.bind(this);
        this.setValue = this.setValue.bind(this);
        this.clearValue = this.clearValue.bind(this);
        this.isFocused = this.isFocused.bind(this);

        this.state = {
            empty: !props.defaultValue,
            focused: false
        };
    }

    componentDidMount() {
        if (!this.props.doNotFocus) {
            this.focus();
        }
    }

    setValue(value) {
        console.log(value);
        console.log(this.refs);
        console.log(this.refs.input);
        if (this.refs.input) {
            this.refs.input.value = value;
        }
    }

    getValue() {
        return this.refs.input ? this.refs.input.value : null;
    }

    clearValue() {
        if (this.refs.input) {
            this.refs.input.value = '';
        }
        this.onChange();
    }

    focus() {
        if (this.refs.input) {
            this.refs.input.focus();
        }
    }

    isFocused() {
        return this.state.focused;
    }

    onFocusHandler() {
        const {doNotScroll} = this.props;
        let inputElem = this.refs.input;
        if (inputElem && !doNotScroll) {
            window.setTimeout(function () {
                inputElem.scrollIntoView();
                const viewClass = document.getElementsByClassName('view');
                if (viewClass && viewClass[0]) {
                    viewClass[0].scrollTop -= 100;
                }
            }, 500);
        }

        this.setState({focused: true});
    }

    onBlurHandler() {
        this.setState({focused: false});
    }

    onChange() {
        if (this.props.onChange) {
            this.props.onChange(this.getValue());
        }

        this.setState({empty: !this.getValue()});
    }

    render() {
        const {placeholder, type, defaultValue, checked, maxLength, minNum, maxNum, searchIcon, textColor, size} = this.props;
        const {empty} = this.state;
        const sizeClass = size === 'small' ? styles.small : null;
        return (
            <div className={styles.inputWrapper + ' ' + sizeClass}>
                <div className={styles.input}>
                    {searchIcon && empty ?
                        <span className={styles.searchIcon + ' icon icon-search'}/>
                        : null
                    }
                    <input ref="input"
                           type={type || "text"}
                           placeholder={placeholder}
                           defaultValue={defaultValue}
                           onChange={this.onChange}
                           onFocus={this.onFocusHandler}
                           onBlur={this.onBlurHandler}
                           maxLength={maxLength}
                           min={minNum}
                           max={maxNum}
                           required
                           style={{color: textColor}}
                    />
                    {/*{!empty && !checked ?*/}
                        {/*<span className={styles.iconCancel + ' ' + 'icon icon-x'} onClick={this.clearValue}/>*/}
                        {/*: null*/}
                    {/*}*/}
                    {!empty && checked ?
                        <span className={'icon icon-check-circle ' + styles.iconCheck}/>
                        : null
                    }
                </div>
            </div>
        );
    }
}