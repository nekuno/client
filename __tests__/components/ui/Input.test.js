import React from 'react';
import Input from '../../../src/js/components/ui/Input/Input.js';
import styles from '../../../src/js/components/ui/Input/Input.scss';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Test Input component', () => {
    const mockFn = jest.fn();
    const input = shallow(<Input placeholder={'Placeholder'} defaultValue={'Default value'} onChange={mockFn}/>);

    it('should be defined', () => {
        expect(Input).toBeDefined();
    });
    it('should render correctly', () => {
        expect(input).toMatchSnapshot();
    });
    it('Input click event', () => {
        input.find(`.${styles.input} input`).simulate('change');
        expect(mockFn).toHaveBeenCalledTimes(1);
    });
});
