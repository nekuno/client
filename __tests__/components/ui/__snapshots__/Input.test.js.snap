// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Input component should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Input
    defaultValue="Default value"
    onChange={[MockFunction]}
    placeholder="Placeholder"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="input"
      >
        <input
          defaultValue="Default value"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Placeholder"
          required={true}
          style={
            Object {
              "color": undefined,
            }
          }
          type="text"
        />
        <span
          className="iconCancel icon icon-x"
          onClick={[Function]}
        />
      </div>,
      "className": "inputWrapper null",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          null,
          <input
            defaultValue="Default value"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Placeholder"
            required={true}
            style={
              Object {
                "color": undefined,
              }
            }
            type="text"
          />,
          <span
            className="iconCancel icon icon-x"
            onClick={[Function]}
          />,
          null,
        ],
        "className": "input",
      },
      "ref": null,
      "rendered": Array [
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "defaultValue": "Default value",
            "max": undefined,
            "maxLength": undefined,
            "min": undefined,
            "onBlur": [Function],
            "onChange": [Function],
            "onFocus": [Function],
            "placeholder": "Placeholder",
            "required": true,
            "style": Object {
              "color": undefined,
            },
            "type": "text",
          },
          "ref": "input",
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "iconCancel icon icon-x",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": "span",
        },
        null,
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="input"
        >
          <input
            defaultValue="Default value"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Placeholder"
            required={true}
            style={
              Object {
                "color": undefined,
              }
            }
            type="text"
          />
          <span
            className="iconCancel icon icon-x"
            onClick={[Function]}
          />
        </div>,
        "className": "inputWrapper null",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            null,
            <input
              defaultValue="Default value"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="Placeholder"
              required={true}
              style={
                Object {
                  "color": undefined,
                }
              }
              type="text"
            />,
            <span
              className="iconCancel icon icon-x"
              onClick={[Function]}
            />,
            null,
          ],
          "className": "input",
        },
        "ref": null,
        "rendered": Array [
          null,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "defaultValue": "Default value",
              "max": undefined,
              "maxLength": undefined,
              "min": undefined,
              "onBlur": [Function],
              "onChange": [Function],
              "onFocus": [Function],
              "placeholder": "Placeholder",
              "required": true,
              "style": Object {
                "color": undefined,
              },
              "type": "text",
            },
            "ref": "input",
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "iconCancel icon icon-x",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": "span",
          },
          null,
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
